### ISSUE TITLE
Python 3.7.0 AttributeError exception  'OrderedDict' not in typing

### ISSUE THREAD

**Issue by aiporre** [beartype/beartype#33]
I got an error Attribute error when I was running python 3.7.0 after 
```
python -c "import beartype"
```
that:

```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/__init__.py", line 36, in <module>
    from beartype._decor.main import beartype
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_decor/main.py", line 231, in <module>
    from beartype._decor._code.codemain import generate_code
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_decor/_code/codemain.py", line 597, in <module>
    from beartype._decor._code._pep.pepcode import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_decor/_code/_pep/pepcode.py", line 37, in <module>
    from beartype._decor._code._pep._pephint import pep_code_check_hint
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_decor/_code/_pep/_pephint.py", line 1722, in <module>
    from beartype._decor._cache.cachetype import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_decor/_cache/cachetype.py", line 35, in <module>
    from beartype._util.hint.utilhinttest import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_util/hint/utilhinttest.py", line 25, in <module>
    from beartype._util.hint.pep.utilhintpeptest import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_util/hint/pep/utilhintpeptest.py", line 23, in <module>
    from beartype._util.hint.data.pep.utilhintdatapep import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_util/hint/data/pep/utilhintdatapep.py", line 27, in <module>
    from beartype._util.hint.data.pep.proposal.utilhintdatapep484 import (
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_util/hint/data/pep/proposal/utilhintdatapep484.py", line 412, in <module>
    _init()
  File "/home/carloberetta/anaconda3/envs/bc/lib/python3.7/site-packages/beartype/_util/hint/data/pep/proposal/utilhintdatapep484.py", line 358, in _init
    typing.OrderedDict,
AttributeError: module 'typing' has no attribute 'OrderedDict'
```

**The workaround was to upgrade to python=3.7.10**

**Comment by aiporre** [comment]
I could try to fix it if you think it useful... :)

**Comment by leycec** [comment]
**Oh noes!** It's horrible. It's also *fascinating* that upgrading the patch version from Python 3.7.0 to 3.7.10 resolved all our appalling inadequacies. I... don't quite know why that would be.

Clearly, we need to support Python 3.7.0 too. I'll grub around in the codebase like a rooting piglet and see if I can't expose the underlying badness.

**Edit:** *Ah, yes.* I see it all so clearly now. Like a [blossoming corpse flower](https://en.wikipedia.org/wiki/Carrion_flower), this issue has been festering undetected for years. We foolishly assumed `typing.OrderedDict` to be unconditionally available under Python >= 3.7. However, in clear violation of sane versioning semantics, `typing.OrderedDict` was only first introduced with Python 3.7.2. :facepalm: 

I'll patch this right up for everyone.

> I could try to fix it if you think its useful... :)

As expected of a dancing unicorn that radiates sparkles named Ariel, you're so nice. I love and adore pull requests, but we've got this one on lockdown. That's an awful pun, 'cause... we're currently in lockdown here in Ontario, Canada. I'll see myself out now.

Thanks again for the detailed report and all the interest, @aiporre! *Resolution incoming.*
